packages/mobile/src/screens/chat-screen/ChatScreen.tsx

```diff
@@ -207,12 +207,16 @@ export const ChatScreen = () => {
   // The chat/chatId selectors will trigger the rerenders necessary.
   const chatFrozenRef = useRef(chat)
 
+  // Initial fetch, but only if messages weren't fetched on app load
   useEffect(() => {
-    if (chatId && (chat?.messagesStatus ?? Status.IDLE) === Status.IDLE) {
-      // Initial fetch
+    if (
+      chatId &&
+      (chat?.messagesStatus ?? Status.IDLE) === Status.IDLE &&
+      chatMessages.length === 0
```

Comment:
possible edge case: chats that are created in the same session with no message sent? Though I _think_ with the idle check it's fine

Reply:
Yeah we'd probably fetch once, which would fail, but then status will be `SUCCESS`. Let me verify.

Reply:
Yeah we fetch once on chat creation, which succeeds with 0 messages, and then on subsequent loads to that chat screen nothing happens due to idle check.